build_tools/mako/benchmark_modules_on_android.py

```diff
@@ -58,12 +58,11 @@ def get_mako_metadata(git_hash, timestamp, benchmark_key) -> str:
 
 def benchmark(module_name, flagfile_name, target) -> str:
   samples = []
-  driver = target.get_driver()
   cmd = [
-      "adb", "shell", "LD_LIBRARY_PATH=/data/local/tmp", "taskset", "80",
-      f"{DEVICE_ROOT}/iree-benchmark-module",
+      "adb", "shell", "LD_LIBRARY_PATH=/data/local/tmp", "taskset",
+      target.taskset, f"{DEVICE_ROOT}/iree-benchmark-module",
```

Comment:
The PR description doesn't seem to match what's happening here. It looks like we used to run everything with `tasket 80` and now we're only doing that for dylib-sync. If true, that's something I'm concerned about because we'd once again be changing the definition of the benchmarks

Reply:
@GMNGeoffrey we are introducing multithreading and single thread benchmarks, for the single thread case we will run in core7 exclusively (for now)  which what `taskset 80` will do, and `taskset F0` for the multithreading

Reply:
That's good, but we also seem to be changing the taskset for, e.g. vulkan. I know this should be GPU bound, but I'm still quite leery of changing what we're measuring. How about instead we have single and multi-threaded benchmarks for each backend/driver?

Reply:
Ah, good catch. This is my mistake.. I somehow thought we used to benchmark with `taskset f0`. Let me change them all back to `taskset 80`.

Reply:
I don't think we do any parallelism at the HAL/runtime side so we can always keep vulkan single threaded .
unpinning HAL from core7 might affect or not the overall performance depends on how expensive these  HAL command buffers and other vulkan-hal stuff cc: @benvanik @antiagainst they know more here :)

Reply:
I think this is a separate problem. This PR basically refactors/exposes more options to `TargetInfo`, and correct single threaded version to use `dylib-sync` driver. I'm happy to address the issue in another PR if this is a concern.

Reply:
I think it still makes sense to keep it pinned for consistency. It's not just specifying single-threaded, it's also pinning to big core